api/api.go

```diff
@@ -207,6 +207,7 @@ func New(requiredUserAgent string, requiredPassword string, cs modules.Consensus
 		router.GET("/renter", api.renterHandlerGET)
 		router.POST("/renter", RequirePassword(api.renterHandlerPOST, requiredPassword))
 		router.GET("/renter/contracts", api.renterContractsHandler)
+		router.GET("/renter/contract/*contractid", api.renterContractDetailHandler)
```

Comment:
it's the convention to have the folder be plural?

e.g. `/renter/contracts/:id` and not `/renter/contract/:id`?

Is it possible / does it make sense to merge the `/renter/contracts` endpoint with the new endpoint? Or can the middleware not handle that?

@RNabel just a warning, because we're committed to maintaining full compatibility with the API for multiple years, this PR is probably going to get a lot of nit-picking, back-and-forth, etc. I think this is a good endpoint to add to the API, but the standards we have for extending the API are very high.